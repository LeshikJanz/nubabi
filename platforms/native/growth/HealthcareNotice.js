import React from 'react';
import { Image } from 'react-native';
import { Box, Card, Text } from '../components';

export const HealthcareNotice = () => {
  const healthcareNoticeIcon = require('core/images/healthcare-notice.png');

  return (
    <Card
      margin={1}
      padding={1}
      flex={1}
      flexDirection="row"
      justifyContent="center"
      alignItems="flex-start"
    >
      <Box marginRight={1} alignSelf="center">
        <Image
          source={healthcareNoticeIcon}
          style={{ width: 60, height: 73 }}
        />
      </Box>
      <Box flex={1} justifyContent="space-around">
        <Text bold lineHeight={20} size={2}>
          Please Note
        </Text>

        <Text color="secondary" lineHeight={17}>
          This information does not replace your health care provider and should
          you be concerned in any way, we recommend that you please consult with
          them or visit your nearest clinic.
        </Text>
      </Box>
    </Card>
  );
};

export default HealthcareNotice;
