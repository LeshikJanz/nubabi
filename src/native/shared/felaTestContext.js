import React from 'react';
import { NativeFelaProvider } from '../components/FelaProvider';
import { createRenderer } from 'fela-native';
import theme from '../../common/themes/defaultTheme';

export const renderer = createRenderer();
const FelaProvider = NativeFelaProvider(renderer, theme);

const felaTestContext = component =>
  <FelaProvider>
    {component}
  </FelaProvider>;

export default felaTestContext;
