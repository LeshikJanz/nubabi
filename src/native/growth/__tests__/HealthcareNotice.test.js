import 'react-native';
import React from 'react';
import { HealthcareNotice } from '../HealthcareNotice';
import renderer from 'react-test-renderer';
import felaTestContext from '../../shared/felaTestContext';

test('it renders correctly', () => {
  const tree = renderer.create(felaTestContext(<HealthcareNotice />)).toJSON();

  expect(tree).toMatchSnapshot();
});
