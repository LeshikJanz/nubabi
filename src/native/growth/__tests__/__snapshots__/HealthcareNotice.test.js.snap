// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#FFFFFF",
      "borderBottomLeftRadius": 4,
      "borderBottomRightRadius": 4,
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopLeftRadius": 4,
      "borderTopRightRadius": 4,
      "borderTopWidth": 0,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginLeft": 16,
      "marginRight": 16,
      "marginTop": 16,
      "paddingBottom": 16,
      "paddingLeft": 16,
      "paddingRight": 16,
      "paddingTop": 16,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "marginRight": 16,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 73,
          "width": 60,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "flex": 1,
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "color": "#454D56",
          "fontFamily": "System",
          "fontSize": 14.791111111111112,
          "fontWeight": "600",
          "lineHeight": 20,
        }
      }
    >
      Please Note
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "color": "#748294",
          "fontFamily": "System",
          "fontSize": 13,
          "lineHeight": 17,
        }
      }
    >
      This information does not replace your health care provider and should you be concerned in any way, we recommend that you please consult with them or visit your nearest clinic.
    </Text>
  </View>
</View>
`;
